app/controllers/collaborators_controller.rb

```diff
@@ -0,0 +1,63 @@
+class CollaboratorsController < ApplicationController
+    before_action :authenticate_user!
+    before_action :set_todo_list, only: [:create]
+    before_action :set_collaborator, only: [:destroy]
+  
+    # POST /todo_lists/:todo_list_id/collaborators
+    def create
+        if @todo_list.user != current_user
+            return render json: { error: 'You are not authorized to perform this action' }, status: :forbidden
+        end
+
+      collaborator = @todo_list.collaborators.build(collaborator_params)
+      authorize collaborator  
+
+      if collaborator.save
+        render json: collaborator, status: :created
+      else
+        render json: { errors: collaborator.errors.full_messages }, status: :unprocessable_entity
+      end
+    rescue ActiveRecord::RecordInvalid => e
+      render json: { errors: [e.record.errors.full_messages.join(', ')] }, status: :unprocessable_entity
+    rescue ActiveRecord::RecordNotUnique
+      render json: { errors: ['Collaborator already exists'] }, status: :unprocessable_entity
+    end
+  
+    # GET /todo_lists/:todo_list_id/collaborators
+  def show
+    collaborators = @todo_list.collaborators.includes(:user)
+    authorize collaborators  
+    render json: collaborators.as_json(only: [:id]), status: :ok
```

Comment:
`collaborators.as_json(only: [:id])` How will it function?

Reply:
through the json output it will fetch only id  as  that parameter is only required but this endpoint is not required , will remove it as it already taken care in todo list

Reply:
its retriving all the collaboratorsf or that specific todolist not a single collaborator but from json it will takes only  id
[
 id:1,
 id:2
]

Reply:
`collaborators.pluck(:id)`